src/utilities/prTitleParser.ts

```diff
@@ -0,0 +1,40 @@
+// TODO: (dunyakirkali) Better naming
+export const parsePRTitle = (title: string, category: string): boolean => {
+  if (category === 'MAJOR') {
+    return true;
+  }
+
+  const semVerTitleRegExp = /bump .* from (?<from>.*) to (?<to>.*)/u;
+  const match = semVerTitleRegExp.exec(title);
+
+  if (match) {
+    const semVerRegExp = /^(?<major>0|[1-9]\d*)\.(?<minor>0|[1-9]\d*)\.(?<patch>0|[1-9]\d*)$/u;
+
+    const matchGroups = match.groups;
+
+    if (matchGroups === undefined || matchGroups.to === undefined || matchGroups.from === undefined) {
+      return true;
+    }
+
+    const from = matchGroups!.from!;
+    const to = matchGroups!.to!;
```

Comment:
@vdvleon something like this? It just looks so ugly

Reply:
Nope. You can never use `!`. I would just write:

```typescript
const from = matchGroups?.from;
const to = matchGroups?.to;

if (from === undefined || to === undefined) {
  return true;
}
```

After this if-statement, TypeScript understands that `to` and `from` are not `undefined`, so in this case a string.